### PR TITLE
Feature/nw23001440/add reset

### PR DESCRIPTION
**Pull request by lanarimarco**
## Description

Added `RESET` opcode implementation. This implementation supports `RESET` just for data definition.
Created `SmeupInterpreterTestCompiled` in order to test the compiled version of smeup unit tests too.

Related to:
- T40_A10_P03D

## Checklist:
- [X] There are tests regarding this feature
- [X] The code follows the Kotlin conventions (run `./gradlew ktlintCheck`)
- [X] The code passes all tests (run `./gradlew check`)
- [ ] There is a specific documentation in the `docs` directory

### CODE REVIEW

File: rpgJavaInterpreter-core/src/test/resources/smeup/T40_A10_P03D.rpgle
Comment (mattiabonardi) [inline rpgJavaInterpreter-core/src/test/resources/smeup/T40_A10_P03D.rpgle:1]: Is file name `T40_A10_P03D` correct ? Or without final `D` character ?

Comment (lanarimarco) [reply rpgJavaInterpreter-core/src/test/resources/smeup/T40_A10_P03D.rpgle:1]: This is the screenshot of MULANGT9, than I think yes.

image

Comment (mattiabonardi) [reply rpgJavaInterpreter-core/src/test/resources/smeup/T40_A10_P03D.rpgle:1]: Perfect!

**Review by mattiabonardi — APPROVED** [review]
Ok

**Pull request merged**